spec/unit/agent_spec.rb

```diff
@@ -85,12 +83,6 @@ def controlled_run(&block)
       @agent.run
     end
 
-    it "should do nothing if already running" do
-      @agent.expects(:running?).returns true
-      AgentTestClient.expects(:new).never
-      @agent.run
-    end
-
```

Comment:
I didn't manage to get such a test working, it seemed like the AgentTestClient was stubbing a bit too much to actually test it. But I felt the code is covered by enough other tests.

But adding a similar one might be good, I likely won't have time for it short term though.